zerver/webhooks/semaphore2/view.py

```diff
@@ -0,0 +1,180 @@
+# Webhooks for external integrations.
+from typing import Any, Dict, List
+
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.webhooks.common import check_send_webhook_message
+from zerver.models import UserProfile
+from urllib.parse import urlparse
+
+# There seems to be a possibility of only 2 types of events : Push or a
```

Comment:
This comment would need to be changed, if indeed there's a `tag` related event. See comment below.